mavros/src/plugins/hil.cpp

```diff
@@ -307,41 +307,39 @@ class HilPlugin : public plugin::PluginBase {
 	 * Message specification: @p https://pixhawk.ethz.ch/mavlink/#HIL_RC_INPUTS_RAW
 	 */
 	void rcin_raw_cb(const mavros_msgs::RCIn::ConstPtr &req) {
-		mavlink::common::msg::HIL_RC_INPUTS_RAW rcin{};
+		mavlink::common::msg::HIL_RC_INPUTS_RAW rcin {};
 
 		constexpr size_t MAX_CHANCNT = 12;
-		size_t channels_count = req->channels.size();
 
-		if (channels_count > MAX_CHANCNT) {
+		if (req->channels.size() > MAX_CHANCNT) {
 			ROS_WARN_THROTTLE_NAMED(60, "hil_rc",
 						"FCU receives %lu RC channels, but HIL_RC_INPUTS_RAW can store %zu",
 						req->channels.size(), MAX_CHANCNT);
-
-			channels_count = MAX_CHANCNT;
 		}
 
+		std::array<uint16_t, MAX_CHANCNT> channels;
+		size_t n = req->channels.size() < MAX_CHANCNT ? : MAX_CHANCNT;
+		std::copy(req->channels.begin(), req->channels.begin() + n, channels.begin());
+		std::fill(channels.begin() + n, channels.end(), UINT16_MAX);
```

Comment:
Remove `if` at line 314, not needed.
line 321 have bug in limit condition (see my test code). Replace with:

```cpp
size_t n = (req->channels.size() < channels.size()) ? req->channels.size() : channels.size();
```

Test code:
```cpp
#include <iostream>
#include <array>
#include <limits>

int main(int argc, char *argv[])
{
	int a = 0, v;
	std::array<uint16_t, 12> arr{};

	if (argc > 1)
		a = atoi(argv[1]);

	v = (a < arr.size()) ? a : arr.size();

	std::cout << "v: " << v << std::endl;

	auto b = arr.begin();
	std::cout << "b: " << b << " b+v: " << b+v << " is end: " << (b+v == arr.end()) << std::endl;

	std::fill(arr.begin() + v, arr.end(), std::numeric_limits<decltype(arr)::value_type>::max());

	std::cout << "arr:";
	for (auto av : arr)
		std::cout << " " << av;
	std::cout << std::endl;

	return 0;
}
```